src/creature.cpp

```diff
@@ -182,9 +182,9 @@ void Creature::onWalk()
 		Direction dir;
 		uint32_t flags = FLAG_IGNOREFIELDDAMAGE;
 		if (getNextStep(dir, flags)) {
-			ReturnValue ret = g_game.internalMoveCreature(this, dir, flags);
+			ReturnValue ret = g_game.internalMoveCreature(getCreature(), dir, flags);
 			if (ret != RETURNVALUE_NOERROR) {
-				if (Player* player = getPlayer()) {
+				if (auto player = getPlayer()) {
```

Comment:
Could this be `const auto`
```suggestion
				if (const auto player = getPlayer()) {
```